.travis.yml

```diff
@@ -1,32 +1,64 @@
-language: rust
-dist: trusty
-sudo: required
+language: python
+sudo: false
+
+python: "3.4"
 
 env:
+  global:
     - PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$HOME/install/syslog-ng/lib/pkgconfig
+    - CMAKE_C_FLAGS=-std=gnu99
+    - LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$HOME/install/syslog-ng/lib"
+    - PATH="$HOME/.local/bin:$PATH"
+  matrix:
+    - RUST_CHANNEL=stable TEST_MODE=everything
+    - RUST_CHANNEL=beta TEST_MODE=everything
+    - RUST_CHANNEL=nightly TEST_MODE=everything TRAVIS_CARGO_NIGHTLY_FEATURE=nightly
 
-# run builds for all the trains (and more)
 matrix:
   include:
-    - rust: stable
-
-before_script:
-  - sudo apt-get update -qq
-  - sudo apt-get install -qq autoconf-archive bison cmake docbook-xsl flex libcap-dev libesmtp-dev libevtlog-dev libgeoip-dev libglib2.0-dev libhiredis-dev libjson0-dev libnet1-dev libpcre3-dev libssl-dev libwrap0-dev openjdk-7-jdk pkg-config python3-all-dev libpython3-dev libpython3-all-dev time xsltproc
-  - cd ..
-  - git clone https://github.com/balabit/syslog-ng.git
-  - cd syslog-ng
-  - ./autogen.sh
-  - ./configure --with-ivykis=internal --prefix=$HOME/install/syslog-ng --disable-python --disable-java --disable-sql --disable-amqp
-  - make -j install
-  - cd ../syslog-ng-rust-modules
-  - mkdir build
-  - cd build
-  - >
-    cmake -DCMAKE_INSTALL_PREFIX=$HOME/install/syslog-ng
-    -DENABLE_REGEX_PARSER=ON
-    -DENABLE_PYTHON_PARSER=ON
-    -DENABLE_ACTIONDB_PARSER=ON
-    -DENABLE_CORRELATION_PARSER=ON ..
+    - env: RUST_CHANNEL=stable TEST_MODE=cmake
+
+git:
+  submodules: false
+
+dist: trusty
+cache:
+  - cargo
+  - ccache
+  - pip
+
+addons:
+  apt:
+    sources:
+    - kubuntu-backports
+    packages:
+    - bison
+    - flex
+    - libcap-dev
+    - libevtlog-dev
+    - libglib2.0-dev
+    - libjson0-dev
+    - libwrap0-dev
+    - pkg-config
+    - cmake
+    - libcurl4-openssl-dev
+    - libelf-dev
+    - libdw-dev
+
+install:
+# copied from https://github.com/dgrunwald/rust-cpython/blob/master/.travis.yml (7c59d554ec6470963b4880345f39b5a543c00496)
+  - pip install 'travis-cargo<0.2'
+  - python -c "import sysconfig; print('\n'.join(map(repr,sorted(sysconfig.get_config_vars().items()))))"
+  - mkdir ~/rust-installer
+  - curl -sL https://static.rust-lang.org/rustup.sh -o ~/rust-installer/rustup.sh
+  - sh ~/rust-installer/rustup.sh --prefix=~/rust --channel=$RUST_CHANNEL -y --disable-sudo
+  - export PATH="$HOME/rust/bin:$PATH"
+  - export PYTHON_LIB=$(python -c "import sysconfig; print(sysconfig.get_config_var('LIBDIR'))")
+  - export LIBRARY_PATH="$LIBRARY_PATH:$PYTHON_LIB"
+  - export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$PYTHON_LIB:$HOME/rust/lib"
+  - rustc -V
+  - bash .travis/install-ivykis.sh
+  - bash .travis/install-syslog-ng.sh
+
 script:
-  - make install
+  - bash .travis/test-${TEST_MODE}.sh
```

Comment:
One way to try to figure out what makes this so slow would be to temporarily try something like this

    .travis/test-${TEST_MODE}.sh 2>&1 |
    while read; do
      echo "$(date "+%Y-%m-%d %H:%M:%S") $REPLY";
    done`